### PR TITLE
chore/initial setup graphql mongoose

### PR DESCRIPTION
**Pull request by melquip**
#### What does this PR do?
- It initializes the server codebase with the most basic setup of express, graphql and mongoose.
- Connects to a real database.
- It has an /api with interactive graphiql endpoint we can test queries on.

#### Description of Task to be completed?
- Set up a good folder structure to properly identify different sections of code in the backend codebase

#### How should this be manually tested?
- To make this work, it will require a valid .env file with working mongodb cluster credentials

#### What are the relevant Trello stories?
- https://trello.com/c/y0PK4oKa

#### Relevant Screenshots (if appropriate)
- None

#### Questions 
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Leelsmuth — COMMENTED** [review]
You might not need to install the additional body-parser package to your application if you are using Express 4.16+

**Comment by melquip** [comment]
Oh, didn't know that. I'll change this tomorrow to use the express.json() instead.

File: middleware/isLoggedIn.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+module.exports = (req, res, next) => {
+  const token = req.authorization;
+  req.isAuth = false;
+  if (!req.authorization) {
+    return next();
+  }
+  let decodedToken;
+  try {
+    decodedToken = jwt.verify(token, 'somesupersecretkey', (decodedToken) => {});
```
Comment (Easybuoy) [inline middleware/isLoggedIn.js]: `somesupersecretkey` should be in environment variables

**Review by Easybuoy — COMMENTED** [review]
This pull request is doing a lot more than initial setup. Try to limit it to the Pull Request scope.

File: middleware/isLoggedIn.js (hunk 1 above)
Comment (melquip) [reply middleware/isLoggedIn.js]: yeah this is something i had from when i made the research, do you want me to remove it and add it later? if not i'll let it be there till we do login

**Review by Leelsmuth — APPROVED** [review]

**Review by Leelsmuth — APPROVED** [review]

**Review by Easybuoy — APPROVED** [review]

**Pull request merged**